### DISCUSSION TITLE
How to pass image to vision model through Workflow Batch Run

### DISCUSSION THREAD

**Discussion by sigvardt** [langgenius/dify#4557] [General]
Hi there

I am looking to batch runs of a Dify workflow app, which includes image files. I am however having the problem that the csv template does not include a variable to pass along the URL of the relevant image. I have tried adding a new column to the csv using relevant variable names, but so far no luck.

**Does anybody know to do batch work using CSV where an input parameter is an image file?** 

![image](https://github.com/langgenius/dify/assets/6000238/a94f3b19-4916-4dc6-b7b7-abd4234ab9be)

![image](https://github.com/langgenius/dify/assets/6000238/152ef6e8-442c-4ed8-ba79-da6fa760d87b)

![image](https://github.com/langgenius/dify/assets/6000238/03dc7ab8-afd2-4e72-9410-dbedcb9b18ab)

**Comment by emadurandal** [comment]
I would like to know that too.
By the way, when I ask LLMs such as ChatGPT4o to analyse the image by giving them the image URL, they say they don't support that.
Therefore, I need to properly connect the URL text in the CSV to the Image Upload (Paste image link) function in Dify. Is that functionality currently available in the current version of Dify? If not, I would like to see it implemented.

**Comment by 11shatteredsky11** [comment]
you just need to pass the url of the image in the csv file, and then use the http node to get the image from the url.
